### PR TITLE
3730 - Port over FedKeyPairGen Tool to StratisBitcoinFullNode 

### PR DESCRIPTION
**Pull request by YakupIpek**
FederationSetup and StratisFederationApp projects migrated to Full Node from FederatedPeg repository. Repository history is also preserved.

### CODE REVIEW

**Comment by fassadlr** [comment]
@YakupIpek please fixed conflicts.

**Comment by YakupIpek** [comment]
@fassadlr  Conflicts are fixed now.

File: src/FederationSetup/GenesisMiner.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        public string MineGenesisBlocks(SmartContractPoAConsensusFactory consensusFactory, string coinbaseText)
+        {
+            var output = new StringBuilder();
+
+            Console.WriteLine("Looking for genesis blocks for the 3 networks, this might take a while.");
+            Console.WriteLine(Environment.NewLine);
+
+            var targets = new Dictionary<uint256, string>
+            {
+                { new uint256("00000fffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"), "-- MainNet network --" },
```
Comment (fassadlr) [inline src/FederationSetup/GenesisMiner.cs]: @YakupIpek can you check as @StratisIain has changed this values recently.

Comment (YakupIpek) [reply src/FederationSetup/GenesisMiner.cs]: It is done.

File: src/FederationSetup/GenesisMiner.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            if (output == null) return;
+            this.output = output;
+        }
+
+        //[Fact]
+        [Fact(Skip = "This is not a test, it is meant to be run upon creating a network")]
+        public void Run_MineGenesis()
+        {
+            var consensusFactory = new SmartContractPoAConsensusFactory();
+            string coinbaseText = "https://www.coindesk.com/apple-co-founder-backs-dorsey-bitcoin-become-webs-currency/";
```
Comment (fassadlr) [inline src/FederationSetup/GenesisMiner.cs]: @YakupIpek I believe this also changed.

File: src/FederationSetup/GenesisMiner.cs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            if (output == null) return;
+            this.output = output;
+        }
+
+        //[Fact]
+        [Fact(Skip = "This is not a test, it is meant to be run upon creating a network")]
+        public void Run_MineGenesis()
+        {
+            var consensusFactory = new SmartContractPoAConsensusFactory();
+            string coinbaseText = "https://github.com/stratisproject/StratisBitcoinFullNode/tree/master/src/Stratis.CirrusD";
```
Comment (fassadlr) [inline src/FederationSetup/GenesisMiner.cs]: The coinbase text is actually different for each Cirrus network.

**Pull request merged**